gnomad/sample_qc/relatedness.py

```diff
@@ -1298,3 +1300,381 @@ def _get_alt_count(locus, gt, is_female):
+        * AC is supposed to be the sum of the alternate alleles in the proband and
+        parents, but we have not implemented this yet because we have multiple trios
+        in one family, in which an allele might be de novo in a parent and
+        transmitted to a child in the dataset.
+
+    locus_expr : hl.expr.LocusExpression
+        Variant's genomic locus.
+    alleles_expr : hl.expr.ArrayExpression
+        Variant alleles.
+    proband_expr : hl.expr.StructExpression
+        Proband genotype details (e.g., DP, GQ, AD, GT, PL).
+    father_expr : hl.expr.StructExpression
+        Father's genotype details (e.g., DP, AD).
+    mother_expr : hl.expr.StructExpression
+        Mother's genotype details (e.g., DP, AD).
+    is_female_expr : hl.expr.BooleanExpression
+        Whether the proband is female.
+    freq_prior_expr : hl.expr.Float64Expression
+        Population frequency prior for the variant.
+    min_pop_prior : float, optional
+        Minimum population frequency prior (default: 100 / 3e7).
+    de_novo_prior : float, optional
+        Prior probability of a de novo mutation (default: 1 / 3e7).
+    min_dp_ratio : float, optional
+        Minimum depth ratio for proband to parents (default: 0.1).
+    min_gq : int, optional
+        Minimum genotype quality for the proband (default: 20).
+    min_proband_ab : float, optional
+        Minimum allele balance for the proband (default: 0.2).
+    max_parent_ab : float, optional
+        Maximum allele balance for parents (default: 0.05).
+    min_de_novo_p : float, optional
+        Minimum de novo probability to pass (default: 0.05).
+    high_conf_dp_ratio : float, optional
+        DP ratio threshold for high confidence (default: 0.2).
+    dp_threshold_snp : int, optional
+        Minimum depth for high-confidence SNPs (default: 10).
+    high_med_conf_ab : float, optional
+        AB threshold for high/medium confidence (default: 0.3).
+    low_conf_ab : float, optional
+        AB threshold for low confidence (default: 0.2).
+    high_conf_p : float, optional
+        P(de novo) threshold for high confidence (default: 0.99).
+    med_conf_p : float, optional
+        P(de novo) threshold for medium confidence (default: 0.5).
+    low_conf_p : float, optional
+        P(de novo) threshold for low confidence (default: 0.2).
+
+    Returns
+    -------
+    hl.expr.StructExpression
+        A struct containing:
+        - `confidence`: Confidence level ("HIGH", "MEDIUM", "LOW", or missing).
+        - `fail`: Boolean indicating if the variant fails any checks.
+        - `fail_reason`: Set of strings with reasons for failure.
+    """
+    p_de_novo = calculate_de_novo_post_prob(
+        proband_expr.PL,
+        father_expr.PL,
+        mother_expr.PL,
+        locus_expr,
+        is_female_expr,
+        freq_prior_expr,
+        min_pop_prior=min_pop_prior,
+        de_novo_prior=de_novo_prior,
+    )
+
+    # Determine genomic context
```

Comment:
rather than running this in this function and in `calculate_de_novo_post_prob`, why not switch the order and run `get_genomic_context` first and pass in the three return values to `calculate_de_novo_post_prob` as function arguments?